### ISSUE TITLE
inconsistent naming for set_input_data

### ISSUE THREAD

**Issue by KrisThielemans** [UCL/STIR#742]
Reconstruction classes use `set_input_data`, while others use `set_input_proj_data_sptr`. I think I'd prefer `set_input_data_sptr`, or even `set_input_sptr`...

**Comment by robbietuk** [comment]
I prefer `set_input_proj_data_sptr `, it is of the same format as `set_additive_proj_data_sptr()` and is very verbose about what it does. `set_input_data_sptr` is okay but `set_input_sptr` is very non-descript.

`set_normalisation_sptr` is the other method that would need standardising to whatever format is chosen (as well as the respecitive `get_` methods.

**Comment by KrisThielemans** [comment]
yes, naming is a bit all over the place once you start looking at it.

I'm more and more leaning towards generic names. The advantage is that it makes generic algorithm design easier. You really want to be able to use EM etc for image-based PVC. In that case, input is an image. Or input can be list mode data. etc.
